### ISSUE TITLE
Error when training on new languages and reproducing results.

### ISSUE THREAD

**Issue by insundaycathy** [oliverguhr/fullstop-deep-punctuation-prediction#9]
Hi Thank you for the great work. 
I followed you guide on preparing my own dataset and training on new languages. 
But during training I got this error:
Unexpected error: <class 'RuntimeError'>
It was right after print the training info and progress bars.

So, then I tried to train on the dataset you provided to see if I've somehow prepared the dataset wrong. And I still got an error in the same place, through it is a different error:
***** Running training *****
  Num examples = 107145
  Num Epochs = 2
  Instantaneous batch size per device = 8
  Total train batch size (w. parallel, distributed & accumulation) = 32
  Gradient Accumulation steps = 1
  Total optimization steps = 6698
  0%|                                                                      | 0/6698 [00:00<?, ?it/s]/home/yidansun/anaconda3/envs/new_env/lib/python3.10/site-packages/torch/nn/parallel/_functions.py:68: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
Unexpected error: <class 'AttributeError'>
  0%|                                                                      | 0/6698 [00:10<?, ?it/s]

As the training is conducted with the transformers Trainer, is why hard to debug the training process. Did you ever encounter similar problems during training?

Thanks in advance.

**Comment by oliverguhr** [comment]
Hi @insundaycathy,
unfortunately this error message could mean anything, so it is hard to say what what went wrong. 
Which GPU do you use? Can you run and post the output of **nvidia-smi** command?

**Comment by insundaycathy** [comment]
I'm using Nvidia gpu, this is the output:
Mon Apr 11 14:40:09 2022
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.47.03    Driver Version: 510.47.03    CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:19:00.0 Off |                  N/A |
| 41%   80C    P2   162W / 350W |   9468MiB / 24576MiB |     49%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce ...  On   | 00000000:1A:00.0 Off |                  N/A |
| 61%   87C    P2   184W / 350W |   9468MiB / 24576MiB |      9%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA GeForce ...  On   | 00000000:67:00.0 Off |                  N/A |
| 49%   87C    P2   190W / 350W |   9468MiB / 24576MiB |     51%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA GeForce ...  On   | 00000000:68:00.0 Off |                  N/A |
| 30%   50C    P8    23W / 350W |      3MiB / 24576MiB |      0%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A   4154651      C   python                           9465MiB |
|    1   N/A  N/A   4160221      C   python                           9465MiB |
|    2   N/A  N/A   4170967      C   python                           9465MiB |
+-----------------------------------------------------------------------------+
Thanks

**Comment by oliverguhr** [comment]
Okay, this looks good. In this case its maybe due to a changed python dependency. Did install all the dependecies in a new virtual environment? And I did not use anaconda, so you can maybe just use plain python venv and pip.

**Comment by insundaycathy** [comment]
May I ask what version of python were you using?

**Comment by oliverguhr** [comment]
I am using python 3.8

**Comment by insundaycathy** [comment]
Thank you so much for all your help. I got the error because I was running with python 3.10. it seem to be a compatibility problem with the package nccl and python 3.10.
When I downgraded to python 3.8 the training both on the orignal dataset and my additonal language went smoothly.

**Comment by nuruddin-jiru** [comment]
hi, @insundaycathy  could you please help me out with training for a new language? I have read the guide but can't understand it much as I am still a beginner sorry. Thanks!
